rust/arrow/src/datatypes.rs

```diff
@@ -751,6 +752,22 @@ impl Schema {
             "fields": self.fields.iter().map(|field| field.to_json()).collect::<Vec<Value>>(),
         })
     }
+
+    /// Create a new schema by applying a projection to this schema's fields
+    pub fn projection(&self, projection: &[usize]) -> Result<Arc<Schema>> {
```

Comment:
Does this belong in `datafusion` as a free function?  It doesn't seem like this will be used within the `arrow` sub-crate?

Reply:
I'm 50/50 on this but I moved it to datafusion for now.